src/api/types.rs

```diff
@@ -1,14 +1,71 @@
+extern crate chrono;
 
-
-#[derive(Queryable, Serialize, Deserialize)]
+#[derive(Queryable, Insertable, Serialize, Deserialize)]
+#[table_name="test_names"]
 pub struct TestName {
-    pub id: i32,
     pub name: String,
 }
 
+#[derive(Queryable, Insertable)]
+#[table_name="artifact_names"]
+pub struct ArtifactName {
+	pub name: String,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="version"]
+pub struct Version {
+	pub id: i32,
+	pub major: String,
+	pub minor: Option<String>,
+	pub patch: Option<String>,
+	pub build: Option<String>,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="test_info"]
+pub struct TestInfo {
+	pub id: i32,
+	pub test_name: String,
+	pub passed: bool,
+	pub artifacts: Vec<String>,
+	pub date: chrono::NaiveDateTime,	// what goes here?
+	pub version: i32,
+	pub link: Option<String>,
+	pub data: Option<Vec<u8>>,	// what goes here?
+}
+
+table! {
+	test_names (name) {
+		name -> Text,
+	}
+}
+
+table! {
+	artifact_names (name) {
+		name -> Text,
+	}
+}
+
+table! {
+	version (id) {
+		id -> Int4,
+		major -> Text,
+		minor -> Nullable<Text>,
+		patch -> Nullable<Text>,
+		build -> Nullable<Text>,
+	}
+}
+
 table! {
-	test_name (id) {
+	test_info (id) {
```

Comment:
I think this table should instead be called `test_run` or something similar. We are not storing data about the test itself, but about the test run.

@vitiral @TheAustinJones @jmcquown